lib/ansible/modules/cloud/linode/linode_v4.py

```diff
@@ -0,0 +1,364 @@
+        U(https://developers.linode.com/api/v4#tag/Tags).
+    required: false
+    type: list
+  root_pass:
+    description:
+      - The password for the root user. If not specified, one will be
+        generated. This generated password will be available in the task
+        success JSON.
+    required: false
+    type: str
+  authorized_keys:
+    description:
+      - A list of SSH public keys to deploy for the root user.
+    required: false
+    type: list
+  state:
+    description:
+      - The desired instance state.
+    type: str
+    choices:
+        - present
+        - absent
+    required: true
+  access_token:
+    description:
+      - The Linode API v4 access token. It may also be specified by exposing
+        the C(LINODE_ACCESS_TOKEN) in the environment. See
+        U(https://developers.linode.com/api/v4#section/Access-and-Authentication).
+    required: true
+"""
+
+EXAMPLES = """
+- name: Create a new Linode.
+  linode_v4:
+    label: new-linode
+    type: g6-nanode-1
+    region: eu-west
+    image: linode/debian9
+    root_pass: passw0rd
+    authorized_keys:
+      - "ssh-rsa ..."
+    state: present
+
+- name: Delete that new Linode.
+  linode_v4:
+    label: new-linode
+    state: absent
+"""
+
+RETURN = """
+root_pass:
+  description: A generated root user password.
+  returned: When a root password is not specified in the task.
+  type: string
+  sample: superSecretPassword
+instance:
+  description: The instance description in JSON serialized form.
+  returned: Always (except delete NOOPs).
+  type: dict
+  sample: {
+    "alerts": {
+      "cpu": 90,
+      "io": 10000,
+      "network_in": 10,
+      "network_out": 10,
+      "transfer_quota": 80
+    },
+    "backups": {
+      "enabled": false,
+      "schedule": {
+        "day": null,
+        "window": null
+      }
+    },
+    "created": "2018-09-26T08:12:33",
+    "group": "Foobar Group",
+    "hypervisor": "kvm",
+    "id": 10480444,
+    "image": "linode/centos7",
+    "ipv4": [
+      "130.132.285.233"
+    ],
+    "ipv6": "2a82:7e00::h03c:46ff:fe04:5cd2/64",
+    "label": "lin-foo",
+    "region": "eu-west",
+    "specs": {
+      "disk": 25600,
+      "memory": 1024,
+      "transfer": 1000,
+      "vcpus": 1
+    },
+    "status": "running",
+    "tags": [],
+    "type": "g6-nanode-1",
+    "updated": "2018-09-26T10:10:14",
+    "watchdog_enabled": true
+  }
+"""
+
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    from linode_api4 import Instance, LinodeClient
+    from linode_api4.errors import (
+        ApiError as LinodeApiError,
+        UnexpectedResponseError
+    )
+    HAS_LINODE_DEPENDENCY = True
+except ImportError:
+    HAS_LINODE_DEPENDENCY = False
+
+try:
+    from ansible.release import __version__ as ansible_version
+except ImportError:
+    ansible_version = 'unknown'
+
+
+ANSIBLE_USER_AGENT = 'Ansible-linode_v4_module/{0}'.format(ansible_version)
```

Comment:
@displague, I hope this is sufficient for the user agent handling that you requested!

Reply:
Commented on in https://github.com/ansible/ansible/pull/46151#pullrequestreview-170698280.